arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,199 @@
+from rtla_schema import (
+    TimerlatInputParams,
+    latency_stats_schema,
+    TimerlatOutput,
+    ErrorOutput,
+)
+
+
+def run_oneshot_cmd(command_list: list[str]) -> tuple[str, subprocess.CompletedProcess]:
+    try:
+        cmd_out = subprocess.check_output(
+            command_list,
+            stderr=subprocess.STDOUT,
+            text=True,
+        )
+    except subprocess.CalledProcessError as err:
+        return "error", ErrorOutput(
+            f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
+        )
+    return "completed", cmd_out
+
+
+class StartTimerlatStep:
+    def __init__(self, exit, finished_early):
+        self.exit = exit
+        self.finished_early = finished_early
+
+    @plugin.signal_handler(
+        id=predefined_schemas.cancel_signal_schema.id,
+        name=predefined_schemas.cancel_signal_schema.display.name,
+        description=predefined_schemas.cancel_signal_schema.display.description,
+        icon=predefined_schemas.cancel_signal_schema.display.icon,
+    )
+    def cancel_step(self, _input: predefined_schemas.cancelInput):
+        # First, let it know that this is the reason it's exiting.
+        self.finished_early = True
+        # Now signal to exit.
+        self.exit.set()
+
+    @plugin.step_with_signals(
+        id="run-timerlat",
+        name="Run RTLA Timerlat",
+        description=(
+            "Runs the RTLA Timerlat data collection and then processes the results "
+            "into a machine-readable format"
+        ),
+        outputs={"success": TimerlatOutput, "error": ErrorOutput},
+        signal_handler_method_names=["cancel_step"],
+        signal_emitters=[],
+        step_object_constructor=lambda: StartTimerlatStep(Event(), False),
+    )
+    def run_timerlat(
+        self,
+        params: TimerlatInputParams,
+    ) -> typing.Tuple[str, typing.Union[TimerlatOutput, ErrorOutput]]:
+
+        timerlat_cmd = ["/usr/bin/rtla", "timerlat", "hist"]
+        timerlat_cmd.extend(params.to_flags())
+
+        try:
+            print("Gathering data... Use Ctrl-C to stop.")
+            proc = subprocess.Popen(
+                timerlat_cmd,
+                start_new_session=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+
+            # Block here, waiting on the cancel signal
+            self.exit.wait(params.duration)
+
+        except subprocess.CalledProcessError as err:
+            return "error", ErrorOutput(
+                f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
+            )
+
+        # Secondary block interrupt is via the KeyboardInterrupt exception.
+        # This enables running the plugin stand-alone without a workflow.
+        except (KeyboardInterrupt, SystemExit):
+            print("\nReceived keyboard interrupt; Stopping data collection.\n")
+            self.finished_early = True
```

Comment:
Should this be two separate `try` blocks, since the `except` clauses apply to different parts of the body?  E.g.,
```suggestion
        try:
            proc = subprocess.Popen(
                timerlat_cmd,
                start_new_session=True,
                stdout=subprocess.PIPE,
                stderr=subprocess.PIPE,
                text=True,
            )
        except subprocess.CalledProcessError as err:
            return "error", ErrorOutput(
                f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
            )

        try:
            # Block here, waiting on the cancel signal
            print("Gathering data... Use Ctrl-C to stop.")
            self.exit.wait(params.duration)

        # Secondary block interrupt is via the KeyboardInterrupt exception.
        # This enables running the plugin stand-alone without a workflow.
        except (KeyboardInterrupt, SystemExit):
            print("\nReceived keyboard interrupt; Stopping data collection.\n")
            self.finished_early = True
```

Reply:
This is fascinating -- I'm suddenly seeing this new "Sign off and commit suggestion" artifact on all of Webb's suggestions. I guess I need to look this up ...
![image](https://github.com/user-attachments/assets/b61a80e5-009b-423e-a674-ed6f5cde3ce3)

Reply:
Yeah, when you comment on a code range (or even reply to such a comment), there's a new button, between the "Preview" tab and the "Heading" format button, which looks like a document with a plus over a minus.  If you click that, it inserts the cited code, and your modifications to it show up as the green...and then the author (or admin) can "simply" "accept" it (the acceptance is simple, but then you've got to pull it back to your local repo if you're going to do anything with it 😛).

The button has got some foibles where sometimes it claims it's not supported, but you can past that by manually typing ` ```suggestion ` instead of your usual triple-back-quote formatting.

Reply:
Hmm. I guess that does make sense. I think when I initially wrote this (note to self: this is the same code that is in the pcp plugin) I interpreted the keyboard interrupt as an exception to the subprocess.